pkg/datastore/firestore/iterator.go

```diff
@@ -33,10 +37,32 @@ func (it *Iterator) Next(dst interface{}) error {
 		}
 		return err
 	}
+
+	// Update last iterated item as last read doc.
+	it.last = doc
+
 	return doc.DataTo(dst)
 }
 
+// Cursor builds a string (in map[string]interface{} format).
+// The cursor contains only values attached with the fields used
+// as ordering fields.
 func (it *Iterator) Cursor() (string, error) {
-	// Note: Cursor function is not needed in Cloud Firestore.
-	return "", datastore.ErrUnimplemented
+	if it.last == nil {
+		return "", datastore.ErrInvalidCursor
+	}
+
+	lastObjData := it.last.Data()
+
+	cursor := make(map[string]interface{})
+	for _, o := range it.orders {
+		val, ok := lastObjData[o.Field]
+		if !ok {
+			return "", datastore.ErrInvalidCursor
+		}
+		cursor[o.Field] = val
+	}
```

Comment:
I think this will not work for a nested field like `SyncState.Status` as [here](https://github.com/pipe-cd/pipe/blob/master/pkg/app/api/grpcapi/web_api.go#L605). But we don't need to support it at the current time so can you add a comment or TODO for the future.

Reply:
Sure, but not sure are we going to support boolean type object as Ordering field 🤔